roles/openshift_logging_fluentd/files/logging_priority_class.yaml

```diff
@@ -0,0 +1,7 @@
+apiVersion: scheduling.k8s.io/v1beta1
+kind: PriorityClass
+metadata:
+  name: cluster-logging
+value: 99999
```

Comment:
What is the default value?  What is the cutoff value for "system-priority level"?
If we raise the priority value, does this mean we will run into the situation where fluentd is not able to be scheduled on a node because other services run at a higher priority?

Reply:
cc @aveshagarwal

Reply:
AFAICT, even if we raise the priority of fluentd to the highest possible level without being system-node-critical or system-cluster-critical, we still run the risk of fluentd being not scheduled or evicted from a node . . .

Reply:
We need this fix for 3.11.  We have not seen this error on 3.10 clusters so I believe we can hold off on any backports.  I have a larger question if we need specific perms to create this resource.

Reply:
> We need this fix for 3.11. We have not seen this error on 3.10 clusters so I believe we can hold off on any backports. 

I dont think 3.11 is an issue here because I noticed that in 3.11 on master branch (https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_logging_fluentd/defaults/main.yml#L5), the logging namespace is `openshift-logging`. 

Also I think 3.10 is also not an issue here because we did not have priority/preemption stuff enabled by default in 3.10. 

But the issue seems to be related to upgrade from 3.10 to 3.11 because in 3.10, the namespace used by logging component is `logging` (https://github.com/openshift/openshift-ansible/blob/release-3.10/roles/openshift_logging_fluentd/defaults/main.yml#L5) where in 3.11/master branch, as i mentioned above, the namespace used is `openshift-logging`.


>I have a larger question if we need specific perms to create this resource.

my guess is that installer should be able to create any priority class at the time of installation.

Reply:
Also why is the value `99999` selected and not `1000000000` which is HighestUserDefinablePriority?

Reply:
Ok, then yes, this is purely related to upgrade to 3.11 - when upgrading from 3.10 to 3.11, we need to migrate _all_ of the logging components/objects from the `logging` namespace to the `openshift-logging` namespace

Reply:
> Ok, then yes, this is purely related to upgrade to 3.11 - when upgrading from 3.10 to 3.11, we need to migrate all of the logging components/objects from the logging namespace to the openshift-logging namespace

Now my question is if we are migrating all logging pods to openshift-logging, then why we are creating this new priority class, why not continue using the existing system-cluster-critical?

Reply:
exactly - we should leave the priority class alone, and remove this PR

Reply:
@richm I spoke to @jwforres and @ewolinetz .  The reason to introduce the priority class are several but the biggest argument against migration is the risk involved in trying to accomplish it in 3.11.  Also we were advised fluent isn't considered 'cluster critical' which means we should probably have our own priority anyway

Reply:
`system-node-critical` (highest priority) is the highest priority class higher than `system-cluster-critical`. All critical (control plane) components including etcd, api, sync-pods, sdn etc are assigned `system-node-critical`. So i am wondering why not assign system-cluster-critical to fluent pods?

Reply:
@aveshagarwal the debate is if the logging stack is as critical as kube-proxy.  i had suggested the option of creating an additional priority (where its not restricted to usage only in openshift-* namespaces) if that would help.

Reply:
> @aveshagarwal the debate is if the logging stack is as critical as kube-proxy. i had suggested the option of creating an additional priority (where its not restricted to usage only in openshift-* namespaces) if that would help.

How are we going to prevent that the newly added priority class specifically for logging related pods will not be used (incorrectly or maliciously) by normal user pods? 

Also, is there any issue if kube-proxy (or similar) is assigned `system-node-critical` and fluentd pods get `system-cluster-critical`? 

My idea is that anything that is critical to running a cluster gets  `system-node-critical` to avoid/reduce cluster downtime or failure. And any thing less critical but still very important for cluster could get  `system-cluster-critical`.

Reply:
@aveshagarwal Is `system-cluster-critical` usable outside of `openshift-*` namespaces?  Is there no mechanism to prevent anyone else from using the priority we create for logging?

Reply:
> @aveshagarwal Is system-cluster-critical usable outside of openshift-* namespaces?

No. Those system priority classes can be used with openshift-* or kube-system.

> Is there no mechanism to prevent anyone else from using the priority we create for logging?

As I mentioned on IRC, resource quota to restrict priority classes is alpha and disabled by default in 3.11. This feature moves to beta in 1.12. So right now, you could look into if it is possible to protect this just one new priority class via rbac. 

However, we are not concerned about someone else using this new priority class, then your current PR is enough.

Reply:
@enj Is it possible to place RBAC rules in place to prevent other users from using a logging priority class?

Reply:
@aveshagarwal is there a way we can restrict our logging priority class to only be used within a specific namespace, similar to how was done for `system-cluster-critical` ?

Reply:
@ewolinetz it would require a patch to carry in openshift like here: https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/plugin/pkg/admission/priority/admission.go#L149-#L157

Reply:
That seems to be more trouble than its worth initially. Unless someone has a strong objection, I think other than updating the value of this priority class to be higher, this PR as is should be enough to resolve the issue seen.